gflownet/envs/crystals/crystal.py

```diff
@@ -536,6 +537,26 @@ def set_state(self, state: List, done: Optional[bool] = False):
             self._get_lattice_parameters_state(state), lattice_parameters_done
         )
 
+        """
+        We synchronize LatticeParameter's lattice system with the one of SpaceGroup
+        (if it was set) or reset it to the default triclinic otherwise. Why this is 
+        needed:
+        1) the first case is necessary for backward sampling, where we start from
+           an arbitrary terminal state, and need to synchronize the LatticeParameter's
+           lattice system to what that state indicates,
+        2) the second case is also necessary in backward sampling, but when we 
+           transition from Stage.LATTICE_PARAMETERS to Stage.SPACE_GROUP. We then need
+           to reset the lattice system to the default triclinic, such that it's
+           source is back to the original one, and corresponds to the source of the
+           whole Crystal environment.
+        """
+        lattice_system = self.space_group.lattice_system
+        if lattice_system != "None":
+            self.lattice_parameters.lattice_system = lattice_system
+        else:
+            self.lattice_parameters.lattice_system = TRICLINIC
+        self.lattice_parameters._set_source()
```

Comment:
Oops... sorry. I hadn't seen that the PR was merged since I first opened the page.

Reply:
I don't think that `self._set_lattice_parameters()` can be called here, since it overwrites `self.lattice_parameters` object (including its state), while we want to keep that in some cases (while in others it doesn't matter) - but correct me if I am wrong. Also, I figured that it would be additional computational overhead to re-create the object every time.

Reply:
You're right that we sometimes want to keep the state. What I was considering was to set the lattice parameters first and then set the state. This way, we wouldn't overwrite the state.
For the overhead, I don't know how bad it would be. I was hoping that it wouldn't be a problem since we already do this object re-creation in the step function but, truth is, I don't know.